src/bot.ts

```diff
@@ -423,7 +423,9 @@ export class DiscordBot {
         }
     }
 
-    public async stop(): Promise<void> {
+    public stop(): void {
+        this.presenceHandler.Stop();
+        this._bot?.destroy();
         this._bot = undefined;
```

Comment:
Nitpick suggestion:
```
if (this._bot) {
    this._bot.destroy();
    this._bot = undefined; 
}
```
since there's a branch buried in the `?.` anyways.